docs/ECOS.md

```diff
@@ -0,0 +1,141 @@
+
+ECO Image Format is aiming at becoming a true extension to an [OCI Image Format Specification](https://github.com/opencontainers/image-spec/blob/master/spec.md). The good news is that OCI Image Spec is structured in an open-ended enough fashion to allow it. The bad news is that it mandates certain aspects of the binary Image Format to be very specific to only containers. The biggest obstacle in from of us is extending [OCI Image Media Types](https://github.com/opencontainers/image-spec/blob/master/media-types.md#oci-image-media-types) to account for binary blobs that are specific to ECI. For example, we would like to have a media type that would correspond to bootable binaries directly (kernels and uikernels) and also initrd binary blobs. Until that happens, we're going to use a workaround that represents these types of objects as binary blobs + metadata wrapped inside of an already supported [Image Layer Filesystem Changeset
+](https://github.com/opencontainers/image-spec/blob/master/layer.md#image-layer-filesystem-changeset).
+
+This approach is not as defeatist as it seems, since by following it we have an immediate benefit of all Docker tooling working with ECIs right away. For example, it becomes possible to produce a Unikernel-based ECI by simply capturing the build of it in a multi-stage Dockerfile with the last stage always being FROM sratch and only copying the kernel binary + metadata into the final image.
+
+### ECI Configuration
+
+ECI Configuration is a json file with the following schema:
+
+    * ACKind (string -- no equivalent in OCI)
+    * ACVersion (string -- no equivalent in OCI -- should be moved to manifest)
+    * Name (string -- no equivalent in OCI -- should be moved to manifest)
+    * Owner (object -- owner.email owner.user is roughtly OCI's author)
+    * Labels (``map<string, string>`` -- OCI's config.label)
+    * Desc (object -- no equivalent in OCI)
+    * ==============
+    * EnableVnc (boolean -- no equivalent in OCI)
+    * VmMode (string -- no equivalent in OCI)
+    * ==============
+    * Resources (struct -- no equivalent in OCI -- describes CPU, memory, storage requirements)
+    * Interfaces (struct -- no equivalent in OCI -- describes image interface requirements)
+    * Images (struct -- OCI's rootfs)
+    * ==============
+    * EventHandlers (this seems deprecated)
+    * Annotations (this seems deprecated)
+    * Permissions (this seems deprecated)
+
+## ECI Distribution Specification
+
+While ECIs are regular, self-contained binary files and can be distributed by any transport (http, ftp, etc.) in certain situations it is advantageuos to define an optimized transport protocol that can be used specifically for ECI distribution.
+
+ECI Distribution Specification will closely follow [OCI Distribution Specification](https://github.com/opencontainers/distribution-spec/blob/master/spec.md) in the hopes to stay compatible with the Docker registry at all times.
```

Comment:
s/in the hopes/with the intent/

Reply:
s/with the Docker registry at all times/with the docker registry specification as much as possible/

Reply:
fixed